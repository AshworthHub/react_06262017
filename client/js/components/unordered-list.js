import React from 'react';
import PropTypes from 'prop-types';

export class UnorderedList extends React.Component {

  static propTypes = {
    listItems: PropTypes.array,
  };

  render() {
    return <ul>
      {this.props.listItems.map((listItem, i) => <li key={i}>
        {listItem}
      </li>)}
    </ul>;
  }

}