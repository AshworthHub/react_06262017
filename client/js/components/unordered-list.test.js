import React from 'react';
import renderer from 'react-test-renderer';

import { UnorderedList } from './unordered-list';

describe('unordered list component', () => {

  test('passing a simple list snapshot', () => {
    const listItems = ['red', 'white', 'blue'];
    const tree = renderer.create(<UnorderedList listItems={listItems} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
