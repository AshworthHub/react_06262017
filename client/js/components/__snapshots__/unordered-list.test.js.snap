// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unordered list component passing a simple list snapshot 1`] = `
<ul>
  <li>
    red
  </li>
  <li>
    white
  </li>
  <li>
    blue
  </li>
</ul>
`;
